pkg/text/truncate.go

```diff
@@ -10,8 +12,11 @@ const (
 	minWidthForEllipsis = 5
 )
 
+var reColorSequence = regexp.MustCompile(`\x1B\[(:?\d+;)*\d+m`)
+
 // DisplayWidth calculates what the rendered width of a string may be
 func DisplayWidth(s string) int {
+	s = reColorSequence.ReplaceAllString(s, "")
```

Comment:
Thank you for the straightforward fix!

Note that an older PR https://github.com/cli/cli/pull/3519/files#diff-7cd07c3defa4823b12484e616fc8cb5f63dc01e5a9df23544634b111a8d20547R15 also proposes ignoring ANSI escape sequences when measuring width of text, but uses a library that does this [without a regular expression](https://github.com/muesli/reflow/blob/9d08d19fb40b1e4b2db5e2731cdb5d318d363ba9/ansi/buffer.go#L21). I would personally vote for that approach since it might be computationally lighter. However, that PR isn't reviewed yet. Please allow us some time to decide how to proceed.